src/lookup/query.ts

```diff
@@ -132,5 +143,20 @@ export async function query(
       )}`,
     );
   }
-  return data;
+
+  if (data.verifiableCredential) {
+    const newVerifiableCredential: (VerifiableCredential & DatasetCore)[] = [];
+    for (let i = 0; i < data.verifiableCredential.length; i += 100) {
+      newVerifiableCredential.push(
+        // eslint-disable-next-line no-await-in-loop
+        ...(await Promise.all(
+          data.verifiableCredential
+            .slice(i, i + 100)
+            .map((vc) => verifiableCredentialToDataset(vc, options)),
+        )),
+      );
+    }
```

Comment:
This is to avoid memory overflow errors from trying to parse too many VCs in parallel.

Granted this shouldn't be an issue with all of the caching and optimisation work that has taken place; I think we should still keep it as there is still a risk of memory errors if, say, millions of VCs are being processed.

I should also note that this doesn't really slow down processing given that we are in a single-threaded environment and most of the operations taking place in the parsing step are synchronous.

Reply:
I've added a comment to the code to reflect this.